src/gc/transaction_level_gc_manager.cpp

```diff
@@ -50,18 +52,19 @@ void TransactionLevelGCManager::Running(const int &thread_id) {
   PL_ASSERT(is_running_ == true);
   uint32_t backoff_shifts = 0;
   while (true) {
-    auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
-    auto max_cid = txn_manager.GetMaxCommittedCid();
+    auto &epoch_manager = concurrency::EpochManagerFactory::GetInstance();
+    
+    auto expired_eid = epoch_manager.GetExpiredEpochId();
     
     // When the DBMS has started working but it never processes any transaction,
-    // we may see max_cid == MAX_CID.
-    if (max_cid == MAX_CID) {
+    // we may see expired_eid == MAX_EID.
```

Comment:
I think I've already explained it. Initially, `expired_eid` is set to `MAX_EID`. So it is possible for the GC to observe the equation.